lib/http/form_data/composite_io.rb

```diff
@@ -29,13 +29,11 @@ def initialize(ios)
       #
       # @return [String, nil]
       def read(length = nil, outbuf = nil)
-        outbuf = outbuf.to_s.clear
-        # buffer in JRuby is sometimes US-ASCII, force to ASCII-8BIT
-        outbuf.force_encoding(Encoding::BINARY)
+        data = outbuf ? outbuf.clear.force_encoding(Encoding::BINARY) : "".b
```

Comment:
I feel neutral, but probably safenave here will look better:

```suggestion
        data = outbuf&.clear&.force_encoding(Encoding::BINARY) || "".b
```

Reply:
Or two lines :joy::

```suggestion
        outbuf &&= outbuf.clear.force_encoding(Encoding::BINARY)
        outbuf ||= "".b
```

Reply:
Just to make it clear - I'm fine to keep the code the way it is now in the PR. Just throwing in ideas how to avoid ternary operator (I usually don't like it unless it's something super simple - like when branches are simple variables or literals).

Reply:
Makes sense about avoiding the ternary operator. How about this?

```rb
data   = outbuf.clear.force_encoding(Encoding::BINARY) if outbuf
data ||= "".b
```

Reply:
Works for me :+1: